### ISSUE TITLE
optimizer set SGD，   The SGD of hyperparameter is not effective

### ISSUE THREAD

**Issue by bluceliuljx** [ultralytics/ultralytics#18839]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

The optimizer was set to SGD to fine tune the model and set some hyperparameters, but these hyperparameters did not take effect, as shown on the left side of the figure below (hyperparameters did not take effect); On the right is when using auto and selecting sgd (normal)

![Image](https://github.com/user-attachments/assets/3622d192-9be7-4fba-b979-a5e3de0c8b09)

![Image](https://github.com/user-attachments/assets/44a71391-660d-4c00-b363-4947588d255d)

![Image](https://github.com/user-attachments/assets/abb0b4af-1bae-48d1-a90c-ebf2a8725381)

### Environment

Ultralytics 8.3.62 🚀 Python-3.8.13 torch-2.0.0+cu117 
CUDA:4 (NVIDIA GeForce RTX 3090, 24260MiB)
CUDA:5 (NVIDIA GeForce RTX 3090, 24260MiB)
CUDA:6 (NVIDIA GeForce RTX 3090, 24260MiB)
CUDA:7 (NVIDIA GeForce RTX 3090, 24260MiB)

### Minimal Reproducible Example

# # SGD
# # lr0: 0.01 # (float) initial learning rate (i.e. SGD=1E-2, Adam=1E-3)
lr0: 0.000003467  # (float) initial learning rate (i.e. SGD=1E-2)    # epoch=100->3.467e-06
lrf: 0.01 # (float) final learning rate (lr0 * lrf)
momentum: 0.937 # (float) SGD momentum/Adam beta1
weight_decay: 0.0005 # (float) optimizer weight decay 5e-4
warmup_epochs: 3.0 # (float) warmup epochs (fractions ok)
warmup_momentum: 0.8 # (float) warmup initial momentum
warmup_bias_lr: 0.1 # (float) warmup initial bias lr

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @bluceliuljx, thank you for bringing this to our attention and for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) where you’ll find helpful usage guides for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/). Many common questions and issues may already be answered there.

If this is a 🐛 Bug Report, as you've described, we request a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us thoroughly understand and debug the issue. Please verify that your hyperparameter configurations are set explicitly in your training script, and ensure your `SGD` optimizer is properly initialized. The code snippet you've shared is helpful; however, including the full relevant training script or configuration files would greatly expedite our diagnostics.

If this pertains to improper behavior during model training, could you also confirm whether you’ve upgraded to the latest version of the `ultralytics` package? You can do this using the following command:

```bash
pip install -U ultralytics
```

This ensures the issue is not due to an outdated version.

## Community Support

Still stuck? Come join our community to get real-time help or insightful discussions:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://reddit.com/r/Ultralytics)

## Environments

To further assist, we recommend testing your workflow in one of the available verified environments, such as:
- **Notebooks** with GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the above badge is green, it indicates all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, verifying that labeling, optimization, and training workflows function as intended.

An Ultralytics engineer will review your report in more detail and assist you as soon as possible. Thank you for your patience and your contribution to improving Ultralytics 💡!

**Comment by Y-T-G** [comment]
Why do you say it didn't take effect? Because the screenshot shows different graphs.
